tools/metricsdocs/metricsdocs.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"kubevirt.io/ssp-operator/internal/operands/metrics"
+	"sort"
+	"strings"
+)
+
+// constant parts of the file
+const (
+	title      = "# SSP Operator metrics\n"
+	background = "This document aims to help users that are not familiar with metrics exposed by the SSP Operator.\n" +
+		"All metrics documented here are auto-generated by the utility tool `tools/metricsdocs` and reflects exactly what is being exposed.\n\n"
+
+	KVSpecificMetrics = "## SSP Operator Metrics List\n"
+
+	opening = title +
+		background +
+		KVSpecificMetrics
+
+	// footer
+	footerHeading = "## Developing new metrics\n"
+	footerContent = "After developing new metrics or changing old ones, please run `make generate-doc` to regenerate this document.\n"
+
+	footer = footerHeading + footerContent
+)
+
+func main() {
+	metricsList := recordRulesDescToMetricList(metrics.RecordRulesDescList)
+	sort.Sort(metricsList)
+	writeToFile(metricsList)
+}
+
+func writeToFile(metricsList metricList) {
+	fmt.Print(opening)
+	metricsList.writeOut()
+	fmt.Print(footer)
+}
+
+type metric struct {
+	name        string
+	description string
+}
+
+func recordRulesDescToMetricList(mdl []metrics.RecordRulesDesc) metricList {
+	res := make([]metric, len(mdl))
+	for i, md := range mdl {
+		res[i] = metricDescriptionToMetric(md)
+	}
+
+	return res
+}
+
+func metricDescriptionToMetric(rrd metrics.RecordRulesDesc) metric {
+	return metric{
+		name:        rrd.Name,
+		description: rrd.Description,
+	}
+}
+
+func (m metric) writeOut() {
+	fmt.Println("###", m.name)
+	fmt.Println(m.description)
+}
+
+type metricList []metric
+
+// Len implements sort.Interface.Len
```

Comment:
I think you can remove this and the next similar comments.

Reply:
Or if you want to check at compile time if `metricList` implements an interface you can add this line:
```go
var _ sort.Interface = metricList{}
```